### PR TITLE
WIP: Project Wally - implementing Mappers

### PR DESCRIPTION
**Pull request by richardjgowers**
This is the prototype atom mapper, which uses rdFMCS a little differently to lomap.

Ben needs to do:
- see project

### CODE REVIEW

File: openfe/setup/atom_mapping/rdfmcs_mapper.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    if mcs_seed:
+        p.InitialSeed = mcs_seed
+    atom_comp = {
+        'any': rdFMCS.AtomCompare.CompareAny,
+        'heavy': rdFMCS.AtomCompare.CompareAnyHeavyAtom,
+        'element': rdFMCS.AtomCompare.CompareElements,
+    }[atom_match.lower()]  # todo catch errors
+    p.AtomTyper = atom_comp
+    p.AtomCompareParameters.MatchValences = atom_match_valences
+    p.AtomCompareParameters.MaxDistance = atom_max_distance
```
Comment (richardjgowers) [inline openfe/setup/atom_mapping/rdfmcs_mapper.py:109]: this is the way to enforce 3d similarity as you are doing the MCS

File: openfe/setup/atom_mapping/rdfmcs_mapper.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    # label the atoms to make mapping the implicit to explicit H version trivial
+    assign_idx(mol1)
+    assign_idx(mol2)
+    # create a copy without hydrogens
+    mol1b = Chem.RemoveHs(mol1)
+    mol2b = Chem.RemoveHs(mol2)
+
+    p = rdFMCS.MCSParameters()
+    if mcs_seed:
+        p.InitialSeed = mcs_seed
```
Comment (richardjgowers) [inline openfe/setup/atom_mapping/rdfmcs_mapper.py:101]: if you can identify a common core ahead of time you can speed up calcs a lot here

Comment (RiesBen) [reply openfe/setup/atom_mapping/rdfmcs_mapper.py]: turned out to be more complicated, as the definition of the MCS is not as liberal as I hoped for. Maybe more setting tweaking is required.

**Comment by RiesBen** [comment]
this pr will continued in #249 and the extra package :)

**Pull request closed without merging**